packages/model-viewer/src/three-components/Model.ts

```diff
@@ -101,41 +102,36 @@ export default class Model extends Object3D {
 
     this.url = url;
 
-    let scene: Scene|null = null;
+    let gltf: ModelViewerGLTFInstance;
 
     try {
-      scene = await new Promise<Scene|null>(async (resolve, reject) => {
-        this[$cancelPendingSourceChange] = () => reject();
-        try {
-          const result = await this.loader.load(url, progressCallback);
-          resolve(result);
-        } catch (error) {
-          reject(error);
-        }
-      });
+      gltf = await new Promise<ModelViewerGLTFInstance>(
+          async (resolve, reject) => {
+            this[$cancelPendingSourceChange] = () => reject();
+            try {
+              const result = await this.loader.load(url, progressCallback);
+              resolve(result);
+            } catch (error) {
+              reject(error);
+            }
+          });
     } catch (error) {
       if (error == null) {
+        // Loading was cancelled, so silently return
         return;
       }
 
       throw error;
     }
 
     this.clear();
-    this[$currentScene] = scene as CacheRetainedScene;
+    this[$currentGLTF] = gltf;
 
-    if (scene != null) {
-      moveChildren(scene, this.modelContainer);
+    if (gltf != null) {
+      moveChildren(gltf.scene, this.modelContainer);
     }
 
-    this.modelContainer.traverse(obj => {
```

Comment:
This was duplicate work, and now consolidated in `ModelViewerGLTFInstance` implementation.